src/App.tsx

```diff
@@ -21,75 +21,139 @@ enum SortType {
-};
+// type ReorderOptions = {
+//   sortType: SortType,
+//   isReversed: boolean,
+// };
 
 // Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+// export function getReorderedGoods(
+//   goods: string[],
+//   { sortType, isReversed }: ReorderOptions,
+// ) {
+//   // To avoid the original array mutation
+//   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+//   // Sort and reverse goods if needed
+//   // eslint-disable-next-line no-console
+//   console.log(sortType, isReversed);
 
-  return visibleGoods;
-}
+//   return visibleGoods;
+// }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  goods: string[],
 };
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const visibleGoods = [...this.state.goods];
+    const { isReversed, sortType } = this.state;
+
+    visibleGoods.sort((a, b): SortType => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return a.length - b.length;
+
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <>
+        <div className="section content">
+          <div className="buttons">
+            <button
+              type="button"
+              className={`button is-info ${this.state.sortType !== SortType.ALPHABET ? 'is-light' : ''}`}
```

Comment:
You can also use [classnames too](https://www.npmjs.com/package/classnames)l for that